### ISSUE TITLE
`diff_against` fails when `HistoricalRecords.excluded_fields` are set

### ISSUE THREAD

**Issue by dave-v** [jazzband/django-simple-history#992]
**Describe the bug**
Since 3.1.0, and specifically https://github.com/jazzband/django-simple-history/pull/776, if you have configured your model with `excluded_fields`, this is not respected when `diff_against` is run.

**To Reproduce**
1. Example model setup:
```py
class TrackedModel(Model):
    field1 = ...
    field2 = ...
    history = HistoricalRecords(excluded_fields=["field2"])
```
2. Create some historical objects
3. `new.diff_against(old)` will raise a `KeyError` on `field2`, but
`new.diff_against(old, excluded_fields=["field2"])` will work

The `KeyError` occurs on https://github.com/jazzband/django-simple-history/blob/106266f6d68f74d2d56b0553a368ca8070d5cbc4/simple_history/models.py#L796 because `old_values` dict defined here https://github.com/jazzband/django-simple-history/blob/106266f6d68f74d2d56b0553a368ca8070d5cbc4/simple_history/models.py#L792 doesn't include `field2`, even though the `fields` set does.

**Expected behavior**
I would expect that the excluded fields as defined on the model would be ignored.
A possible one-line fix is to change: https://github.com/jazzband/django-simple-history/blob/106266f6d68f74d2d56b0553a368ca8070d5cbc4/simple_history/models.py#L780 like so:
```diff
-            excluded_fields = set()
+            excluded_fields = set(old_history._history_excluded_fields)
```
though that's probably not the most robust solution.

**Screenshots**

**Environment (please complete the following information):**
 - OS: Windows 10
 - Browser (if applicable): N/A
 - Django Simple History Version: 3.1.1
 - Django Version: 3.2.13
 - Database Version: PostgreSQL 13.6

**Additional context**
https://github.com/jazzband/django-simple-history/pull/923 seems to be related, but doesn't fix this issue.

**Comment by alexanderoskin** [comment]
The same issue

**Comment by zhaixiuyan** [comment]
me too!

**Comment by userlocalhost** [comment]
me either.

**Comment by semente** [comment]
is there any workaround?

**Comment by dmartin** [comment]
Possible duplicate of (resolved) https://github.com/jazzband/django-simple-history/issues/988?

**Comment by ddabble** [comment]
I'm able to reproduce the error before applying the fix to the mentioned #988, but not after, so I'm assuming that the issue has been resolved. Please reopen if it turns out that's not the case after all 🙂
